docs/codebase.md

```diff
@@ -0,0 +1,84 @@
+Codebase Overview
```

Comment:
I'd like to add a dependency graph of the Table Render / Component Tree section.
I'd also like to add some detail on the Redux migration / how this will look after that.